std/math/emulated/field_reduce.go

```diff
@@ -0,0 +1,53 @@
+package emulated
+
+// ReduceWidth returns an element reduced by the modulus and constrained to have
+// same length as the modulus. The output element has the same width as the
+// modulus but may up to twice larger than the modulus).
+//
+// Does not mutate the input.
+//
+// In cases where the canonical representation of the element is required, use
+// [Field.ReduceStrict].
+func (f *Field[T]) Reduce(a *Element[T]) *Element[T] {
+	ret := f.reduce(a, false)
+	return ret
+}
+
+func (f *Field[T]) reduce(a *Element[T], strict bool) *Element[T] {
+	f.enforceWidthConditional(a)
+	if strict && a.modReduced {
```

Comment:
why do we check if the `strict` flag is provided? 
if `modReduced` is set, both in Reduce and Reduce strict, we should return `a`

Reply:
Fixed.